test/suites/integration/webhook_test.go

```diff
@@ -0,0 +1,170 @@
+
+			Expect(len(provisioner.Spec.Requirements)).To(Equal(2))
+			Expect(provisioner.Spec.Requirements).To(ContainElement(v1.NodeSelectorRequirement{
+				Key:      v1alpha5.LabelCapacityType,
+				Operator: v1.NodeSelectorOpIn,
+				Values:   []string{awsv1alpha1.CapacityTypeOnDemand},
+			}))
+			Expect(provisioner.Spec.Requirements).To(ContainElement(v1.NodeSelectorRequirement{
+				Key:      v1.LabelArchStable,
+				Operator: v1.NodeSelectorOpIn,
+				Values:   []string{"amd64"},
+			}))
+		})
+	})
+	Context("Validating Webhook", func() {
+		It("should deny the request when no providerRef is set", func() {
+			provisioner := test.Provisioner()
+			Expect(env.Client.Create(env, provisioner)).ToNot(Succeed())
+		})
+		It("should deny the request when a restricted label is used in labels (karpenter.sh/provisioner-name)", func() {
+			provider := test.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{AWS: awsv1alpha1.AWS{
+				SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": env.ClusterName},
+				SubnetSelector:        map[string]string{"karpenter.sh/discovery": env.ClusterName},
+			}})
+			provisioner := test.Provisioner(test.ProvisionerOptions{
+				ProviderRef: &v1alpha5.ProviderRef{Name: provider.Name},
+				Labels: map[string]string{
+					v1alpha5.ProvisionerNameLabelKey: "my-custom-provisioner",
+				},
+			})
+			Expect(env.Client.Create(env, provisioner)).ToNot(Succeed())
+		})
+		It("should deny the request when a restricted label is used in labels (kubernetes.io/custom-label)", func() {
+			provider := test.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{AWS: awsv1alpha1.AWS{
+				SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": env.ClusterName},
+				SubnetSelector:        map[string]string{"karpenter.sh/discovery": env.ClusterName},
+			}})
+			provisioner := test.Provisioner(test.ProvisionerOptions{
+				ProviderRef: &v1alpha5.ProviderRef{Name: provider.Name},
+				Labels: map[string]string{
+					"kubernetes.io/custom-label": "custom-value",
+				},
+			})
+			Expect(env.Client.Create(env, provisioner)).ToNot(Succeed())
+		})
+		It("should deny the request when a requirement references a restricted label (karpenter.sh/provisioner-name)", func() {
+			provider := test.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{AWS: awsv1alpha1.AWS{
+				SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": env.ClusterName},
+				SubnetSelector:        map[string]string{"karpenter.sh/discovery": env.ClusterName},
+			}})
+			provisioner := test.Provisioner(test.ProvisionerOptions{
+				ProviderRef: &v1alpha5.ProviderRef{Name: provider.Name},
+				Requirements: []v1.NodeSelectorRequirement{
+					{
+						Key:      v1alpha5.ProvisionerNameLabelKey,
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{"default"},
+					},
+				},
+			})
+			Expect(env.Client.Create(env, provisioner)).ToNot(Succeed())
+		})
+		It("should deny the request when a requirement uses an unknown operator", func() {
```

Comment:
Can you also add a test for when a requirement has an operator but no values? 

https://github.com/aws/karpenter/blob/55925ed9d4d8d2c13dab89fced5f7882e83e61a9/pkg/apis/provisioning/v1alpha5/provisioner_validation.go#L285